solidity/deploy/08_deploy_maintainer_proxy.ts

```diff
@@ -0,0 +1,33 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts } = hre
+  const { deploy } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const Bridge = await deployments.get("Bridge")
+  const ReimbursementPool = await deployments.get("ReimbursementPool")
+
+  const MaintainerProxy = await deploy("MaintainerProxy", {
+    contract:
+      deployments.getNetworkName() === "hardhat"
+        ? "MaintainerProxyStub"
+        : undefined,
+    from: deployer,
+    args: [Bridge.address, ReimbursementPool.address],
```

Comment:
Just to ensure, `Bridge.address` will point to the proxy, not to the implementation? We need to test this contract calling via proxy. cc @nkuba to confirm

Reply:
@nkuba do we want to replace `Bridge.address` with `BridgeProxyAdminWithDeputy.address` from [BridgeProxyAdminWithDeputy script](https://github.com/keep-network/tbtc-v2/blob/main/solidity/deploy/12_deploy_proxy_admin_with_deputy.ts)?

Reply:
> Just to ensure, Bridge.address will point to the proxy, not to the implementation?

Yes, it should resolve to the Bridge proxy address.

> @nkuba do we want to replace Bridge.address with BridgeProxyAdminWithDeputy.address from BridgeProxyAdminWithDeputy script?

No, this should still be `Bridge.address`. The `BridgeProxyAdminWithDeputy` is used for `Bridge` upgrades, we almost never will use it.

> We need to test this contract calling via proxy.

Yes, we need to test it very much! Maybe in https://github.com/keep-network/tbtc-v2/blob/main/solidity/test/bridge/Bridge.Deployment.test.ts?

Reply:
> The BridgeProxyAdminWithDeputy is used for Bridge upgrades, we almost never will use it.

> Yes, we need to test it very much!

Sorry, not sure if I follow. We have [a bunch of tests](https://github.com/keep-network/tbtc-v2/blob/maintainer-proxy/solidity/test/maintainer/Maintainer.test.ts) that call Maintainer Proxy contract which calls Bridge contract. 
Also, we have tests in `Bridge.Deployment.test.ts` as you pointed out that deals with the Bridge proxy thing. Maintainer proxy only cares about the Bridge contract, so as long as the upgradability worked fine (which as far as I understand has its own separate track) then we should be good. Unless what you mean is that we should test calling Maintainer Proxy contract _after_ the Bridge was upgraded? If yes, wdyt about doing it in a seperate PR, cause imo it is closer to the upgradability functionality rather than the MaintainerProxy itself?